source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -64,13 +89,39 @@ bool LocalRateLimiterImpl::requestAllowed() const {
     synchronizer_.syncPoint("allowed_pre_cas");
 
     // Loop while the weak CAS fails trying to subtract 1 from expected.
-  } while (!tokens_.compare_exchange_weak(expected_tokens, expected_tokens - 1,
-                                          std::memory_order_relaxed));
+  } while (!tokens.tokens.compare_exchange_weak(expected_tokens, expected_tokens - 1,
+                                                std::memory_order_relaxed));
 
   // We successfully decremented the counter by 1.
   return true;
 }
 
+bool LocalRateLimiterImpl::requestAllowed() const { return requestAllowedHelper(tokens_); }
+
+void LocalRateLimiterImpl::fillDescriptor(
+    const std::vector<Envoy::RateLimit::LocalDescriptor>& descriptors) {
+  for (const auto& descriptor : descriptors) {
+    std::unique_ptr<Token> token = std::make_unique<Token>();
+    if (descriptor.token_bucket_.fill_interval_.count() % fill_interval_.count() != 0) {
+      throw EnvoyException(
+          "local rate descriptor limit is not a  multiple of token bucket fill timer");
```

Comment:
```suggestion
          "local rate descriptor limit is not a multiple of token bucket fill timer");
```

Reply:
Also needs an EXPECT_THROW_WITH_MESSAGE test

Reply:
Already added a test with EXPECT_THROW_WITH_MESSAGE in local_ratelimit_test